Core/include/Acts/EventData/ParticleHypothesis.hpp

```diff
@@ -0,0 +1,127 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/PdgParticle.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/Charge.hpp"
+#include "Acts/EventData/GenericParticleHypothesis.hpp"
+
+namespace Acts {
+
+// TODO In principle the factory methods could provide a reference to a static
+// instance which would avoid copying the particle hypothesis and potentially
+// save some memory. But constexpr+static seems to require C++2b extension.
+
+class SinglyChargedParticleHypothesis
+    : public GenericParticleHypothesis<SinglyCharged> {
```

Comment:
Private inheritance is maybe better? Are we holding these as `GenericParticleHypothesis` or as template parameters anyway?

Reply:
I think it will only be used as a template param the inheritance is just used the get the methods into this class and to have the right factory methods.

private inheritance would shadow the public methods of the parents no?

Reply:
Yeah private inheritance would make all inherited methods private. And also would forbid pointer/reference conversion to base-class, but that should not be too relevant here.

Reply:
Yeah, right. Nevermind then.